### PR TITLE
Add GH action to close stale PRs

### LINKED ISSUE #7151: Enable GitHub Action to Close Inactive PRs
### Area for Improvement

Maintaining a clean and manageable PR queue is essential for project's maintainability. This proposal is to add a workflow to automatically close inactive PRs after a specified period. This will help declutter the PR queue and ensure that active contributions receive attention. A PR could land in this state due to many reasons, such as the submitter deciding not to continue working on the PR, changes in project direction making the PR irrelevant, or if the changes were included in another PR.

### Observed behavior

_No response_

### Desired behavior

Leverage the https://github.com/actions/stale GitHub action to automatically close PRs after 90 days of inactivity, with a notification period of 7 days. This will provide contributors with a reasonable window to address any outstanding feedback or issues before the PR is automatically closed. We can adjust this timeframe in the future based on the feedback and learnings. We can also consider PRs labeled with certain labels to be excluded from this workflow if we see the need for this in the future: https://github.com/actions/stale?tab=readme-ov-file#exempt-pr-labels.

### Proposed Fix

_No response_

### rad Version

Not applicable

### Operating system

_No response_

### Additional context

_No response_

### Would you like to support us?

- [ ] Yes, I would like to support you

[AB#11137](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/11137)

### PR DESCRIPTION
**Pull request by kachawla**
# Description

To declutter the PR queue and ensure that active contributions receive attention, enabling https://github.com/actions/stale github action to automatically close stale PRs after 90 days of inactivity, with a notification period of 7 days.

This will provide contributors with a reasonable window to address any outstanding feedback or issues before the PR is automatically closed. We can adjust this timeframe in the future based on the feedback and learnings. We can also consider PRs labeled with certain labels to be excluded from this workflow if we see the need for this in the future: https://github.com/actions/stale?tab=readme-ov-file#exempt-pr-labels.

## Type of change

- This pull request adds or changes features of Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: https://github.com/radius-project/radius/issues/7151

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Contributing Pull Requests
```
Comment (kachawla) [inline CONTRIBUTING.md:1]: @AaronCrawfis Copied most of it from the Radius repo contributions guide. Let me know what you think.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+- For very minor changes like correcting a typo feel free to just send a pull request without any ceremony. Otherwise ... 
+- Please start by [choosing an existing issue](https://github.com/radius-project/samples/issues), or [opening an issue](https://github.com/radius-project/samples/issues/new/choose) to work on.
+- The maintainers will respond to your issue, please work with the maintainers to ensure that what you're doing is in scope for the project before writing any code.
+- If you have any doubt whether a contribution would be valuable, feel free to ask.
+
+We the maintainers have discretion over what features and pull requests we accept. Please understand that we are responsible for the long-term support and maintenance of Radius, and so we sometimes need to make hard decisions to limit the scope. For another perspective on this, we really like this [article](https://www.igvita.com/2011/12/19/dont-push-your-pull-requests/).
+
+## Sending a pull request
+Please submit pull requests using a forked repo and open pull requests against the `edge` branch (the default) unless otherwise instructed.
```
Comment (kachawla) [inline CONTRIBUTING.md]: @AaronCrawfis should this be `edge` or the latest release branch?

File: CONTRIBUTING.md (hunk 1 above)
Comment (AaronCrawfis) [reply CONTRIBUTING.md:1]: Thanks for noticing that we're missing a CONTRIBUTING.md file! We'll definitely need one. Could you actually just make this file a pointer to https://docs.radapp.io/community/contributing/docs/ ?

We have most of the contributing information there.

Thanks!!

Comment (AaronCrawfis) [reply CONTRIBUTING.md:1]: Thought I was in the docs repo, please disregard 😆

File: CONTRIBUTING.md (hunk 2 above)
Comment (AaronCrawfis) [inline CONTRIBUTING.md]:
```suggestion

Please submit pull requests using a forked repo and open pull requests against:

- The default, versioned branch (`v0.30` for example) if you are adding/fixing a sample and it is compatible with the latest release of Radius.
- The `edge` branch if you are contributing a sample that requires a feature that will be available in the next release of Radius.
```

Hmmm it's a little nuanced based on what you're trying to do. Lemme see if I can take a crack at it above

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+You can "resolve" comments on your pull request when you've addressed the feedback: either through discussion or through making a code change. As the contributor of the pull-request feel free to mark comments as resolved when you feel like you've done a reasonable job addressing the feedback.
+
+If you are the code reviewer, it's your responsibility to follow up (politely) if you feel your feedback has not been addressed adequately.
+
+### Participating in code review
+
+We welcome **any contributor or community member** to engage with **any pull request** on our repository. Feel free to make suggestions for improvements and ask questions that are relevant. If you're asking questions for your learning, please make it clear that your questions are "non-blocking" for the pull request.
+
+See the [code reviewing documentation](../contributing-code/contributing-code-reviewing/README.md) for guidance on code reviewing.
```
Comment (AaronCrawfis) [inline CONTRIBUTING.md]: This link will be broken, let's update it to an fully qualified URL to the radius repo

File: CONTRIBUTING.md (hunk 2 above)
Comment (kachawla) [reply CONTRIBUTING.md]: Makes sense, thank you! :)

File: CONTRIBUTING.md (hunk 3 above)
Comment (kachawla) [reply CONTRIBUTING.md]: Good catch

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
```
Comment (kachawla) [inline CONTRIBUTING.md]:
```suggestion
See the [code reviewing documentation](https://github.com/radius-project/radius/blob/main/docs/contributing/contributing-code/contributing-code-reviewing/README.md) for guidance on code reviewing.
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (kachawla) [reply CONTRIBUTING.md]: Updated. Thanks for reviewing! Let me know if anything else looks off.

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**